src/shillelagh/fields.py

```diff
@@ -469,8 +506,8 @@ def parse(self, value: Optional[str]) -> Optional[datetime.datetime]:
             return None
 
         try:
-            timestamp = dateutil.parser.parse(value)
-        except dateutil.parser.ParserError:
+            timestamp = datetime.datetime.fromisoformat(value)
```

Comment:
for this method, it might be safer to use `dateutil.parser.isoparse` based on:
>Caution This does not support parsing arbitrary ISO 8601 strings - it is only intended as the inverse operation of [datetime.isoformat()](https://docs.python.org/3/library/datetime.html#datetime.datetime.isoformat). A more full-featured ISO 8601 parser, dateutil.parser.isoparse is available in the third-party package [dateutil](https://dateutil.readthedocs.io/en/stable/parser.html#dateutil.parser.isoparse).

Reply:
Yeah, I saw that... I think there's some value in having these fields (`ISODate`, `ISODateTime`, `ISOTime`) be fast and strict, since they are used when the data moves through the SQLite layer (the DB API driver serializes temporal data using these fields, passes the data to `apsw`, and when it comes out we deserialize using the same fields):

https://github.com/betodealmeida/shillelagh/blob/adff3f8e6c25c7c4257a1211f2425b19e771d827/src/shillelagh/backends/apsw/vt.py#L110-L153

Adapters can then use the `StringDate`, `StringDateTime`, and `StringTime` fields for their columns, in order to use `dateutil.parser.isoparse`.

Reply:
"strict" isn't really the right word here though since that built in parser is actually stricter than even the ISO standard (ie it will reject strings that are valid according to the standard). perhaps add a "fast" flag to the field that determines which iso parser is used? 

I think otherwise there will be people confused with the iso parser failing on valid iso strings

Reply:
Good point. Let's use `dateutil.parse.isoparse` in the `ISO*` fields, and I'll create `FastISO*` fields for this.

Also, I'm working on 1.1... no big changes planned (other than `DROP TABLE` that was implemented last week) and no breaking changes expected (it is 1.x, after all). But let me know if there's more stuff that you need in Shillelagh, since I'll probably take a break after that.

Reply:
I think for date you can use the built in as there is only one format AFAIK